BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/ProgrammaticSliderViewController.swift

```diff
@@ -0,0 +1,62 @@
+import UIKit
+import AVFoundation
+
+class ProgrammaticSliderViewController: UIViewController {
+    let slider = UISlider()
+    var imageView : UIImageView!
+    var songPlayer: AVAudioPlayer?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = UIColor.white
+        setupSlider()
+        setupBackground()
+        layoutConstraints()
+        DispatchQueue.global(qos: .userInitiated).async { self.playSong() }
+        }
+    func setupSlider() {
+        view.addSubview(slider)
+        slider.addTarget(self, action: #selector(self.changeSliderValue(_:)), for: .valueChanged)
+        slider.minimumValue = 0.0
+        slider.maximumValue = 1.0
+        slider.value = 1
+        slider.maximumTrackTintColor = UIColor.black
+        slider.thumbTintColor = UIColor.blue
+        slider.isContinuous = true
+    }
+    func layoutConstraints() {
+        slider.translatesAutoresizingMaskIntoConstraints = false
+        slider.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        slider.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -100).isActive = true
+        slider.widthAnchor.constraint(equalToConstant: 250).isActive = true
+        slider.heightAnchor.constraint(equalToConstant: 50).isActive = true
+    }
+    @objc func changeSliderValue(_ sender: UISlider) {
+        print("value is \(sender.value)");
+        songPlayer?.volume = slider.value
+    }
+    func setupBackground(){
+        let background = UIImage(named: "moonLake")
+        imageView = UIImageView(frame: view.bounds)
```

Comment:
Nice. Without looking this up, answer this typical interview question: why `view.bounds` and not `view.frame`?

Reply:
If I remember correctly, the bounds has to do with the outer edges of the view rectangle, where as the frame deals with the location of the view rectangle.

Reply:
Almost! Bounds is in reference to itself, while frame is in reference with its superview.

![bounds size vs frame size](https://user-images.githubusercontent.com/8583018/53933255-76c7e180-406b-11e9-8f9d-f98b5be981f1.png)